<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InstanceSettingsPage</class>
 <widget class="QWidget" name="InstanceSettingsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>691</width>
    <height>581</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QCommandLinkButton" name="openGlobalJavaSettingsButton">
     <property name="text">
      <string>Open Global Settings</string>
     </property>
     <property name="description">
      <string>The settings here are overrides for global settings.</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="settingsTabs">
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="minecraftTab">
      <attribute name="title">
       <string notr="true">Java</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="javaSettingsGroupBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Java insta&amp;llation</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0" colspan="3">
           <widget class="QLineEdit" name="javaPathTextBox"/>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="javaDetectBtn">
            <property name="text">
             <string>Auto-detect...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="javaBrowseBtn">
            <property name="text">
             <string>Browse...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="javaTestBtn">
            <property name="text">
             <string>Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="memoryGroupBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Memor&amp;y</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="labelMinMem">
            <property name="text">
             <string>Minimum memory allocation:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="maxMemSpinBox">
            <property name="toolTip">
             <string>The maximum amount of memory Minecraft is allowed to use.</string>
            </property>
            <property name="suffix">
             <string notr="true"> MB</string>
            </property>
            <property name="minimum">
             <number>128</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
            <property name="singleStep">
             <number>128</number>
            </property>
            <property name="value">
             <number>1024</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="minMemSpinBox">
            <property name="toolTip">
             <string>The amount of memory Minecraft is started with.</string>
            </property>
            <property name="suffix">
             <string notr="true"> MB</string>
            </property>
            <property name="minimum">
             <number>128</number>
            </property>
            <property name="maximum">
             <number>65536</number>
            </property>
            <property name="singleStep">
             <number>128</number>
            </property>
            <property name="value">
             <number>256</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="permGenSpinBox">
            <property name="toolTip">
             <string>The amount of memory available to store loaded Java classes.</string>
            </property>
            <property name="suffix">
             <string notr="true"> MB</string>
            </property>
            <property name="minimum">
             <number>64</number>
            </property>
            <property name="maximum">
             <number>999999999</number>
            </property>
            <property name="singleStep">
             <number>8</number>
            </property>
            <property name="value">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="labelPermGen">
            <property name="text">
             <string notr="true">PermGen:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelMaxMem">
            <property name="text">
             <string>Maximum memory allocation:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLabel" name="labelPermgenNote">
            <property name="text">
             <string>Note: Permgen is set automatically by Java 8 and later</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="javaArgumentsGroupBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Java argumen&amp;ts</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="1" column="1">
           <widget class="QPlainTextEdit" name="jvmArgsTextBox"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="javaTab">
      <attribute name="title">
       <string>Game windows</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="windowSizeGroupBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Game Window</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="maximizedCheckBox">
            <property name="text">
             <string>Start Minecraft maximized?</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayoutWindowSize">
            <item row="1" column="0">
             <widget class="QLabel" name="labelWindowHeight">
              <property name="text">
               <string>Window height:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="labelWindowWidth">
              <property name="text">
               <string>Window width:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="windowWidthSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65536</number>
              </property>
              <property name="singleStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>854</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="windowHeightSpinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>65536</number>
              </property>
              <property name="value">
               <number>480</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="consoleSettingsBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Conso&amp;le Settings</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QCheckBox" name="showConsoleCheck">
            <property name="text">
             <string>Show console while the game is running?</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="autoCloseConsoleCheck">
            <property name="text">
             <string>Automatically close console when the game quits?</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="showConsoleErrorCheck">
            <property name="text">
             <string>Show console when the game crashes?</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacerMinecraft_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>88</width>
           <height>125</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Custom commands</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="CustomCommands" name="customCommands" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="workaroundsPage">
      <attribute name="title">
       <string>Workarounds</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QGroupBox" name="nativeWorkaroundsGroupBox">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="title">
          <string>Native libraries</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QCheckBox" name="useNativeGLFWCheck">
            <property name="text">
             <string>Use system installation of GLFW</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="useNativeOpenALCheck">
            <property name="text">
             <string>Use system installation of OpenAL</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CustomCommands</class>
   <extends>QWidget</extends>
   <header>widgets/CustomCommands.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>openGlobalJavaSettingsButton</tabstop>
  <tabstop>settingsTabs</tabstop>
  <tabstop>javaSettingsGroupBox</tabstop>
  <tabstop>javaPathTextBox</tabstop>
  <tabstop>javaDetectBtn</tabstop>
  <tabstop>javaBrowseBtn</tabstop>
  <tabstop>javaTestBtn</tabstop>
  <tabstop>memoryGroupBox</tabstop>
  <tabstop>minMemSpinBox</tabstop>
  <tabstop>maxMemSpinBox</tabstop>
  <tabstop>permGenSpinBox</tabstop>
  <tabstop>javaArgumentsGroupBox</tabstop>
  <tabstop>jvmArgsTextBox</tabstop>
  <tabstop>windowSizeGroupBox</tabstop>
  <tabstop>maximizedCheckBox</tabstop>
  <tabstop>windowWidthSpinBox</tabstop>
  <tabstop>windowHeightSpinBox</tabstop>
  <tabstop>consoleSettingsBox</tabstop>
  <tabstop>showConsoleCheck</tabstop>
  <tabstop>autoCloseConsoleCheck</tabstop>
  <tabstop>showConsoleErrorCheck</tabstop>
  <tabstop>nativeWorkaroundsGroupBox</tabstop>
  <tabstop>useNativeGLFWCheck</tabstop>
  <tabstop>useNativeOpenALCheck</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
